### PR TITLE
passive income

### LINKED ISSUE #48: add other resource buildins
stone
wood

### PR DESCRIPTION
**Pull request by hoolek77**

### CODE REVIEW

File: src/app/game.ts [hunk 1]
```diff
@@ -328,8 +326,16 @@ export class Game {
...
   }
 
-  private handleGoldmineWasBuilt(goldmine: Goldmine) {
-    this.changeGoldAmount(this.getGoldAmount() + goldmine.goldProduction)
+  handleGoldmineWasBuild(goldmine: Goldmine) {
+    this.changeGoldAmount(this.getGoldAmount() + goldmine.getProduction())
+  }
+
+  handleQuarryeWasBuild(quarry: Quarry) {
+    this.changeGoldAmount(this.getGoldAmount() + quarry.getProduction())
```
Comment (bartoszbialecki) [inline src/app/game.ts]: Here should be this.getStoneAmount()

File: src/app/game.ts [hunk 2]
```diff
@@ -328,8 +326,16 @@ export class Game {
...
+  }
+
+  handleSawmillWasBuild(sawmill: Sawmill) {
+    this.changeGoldAmount(this.getGoldAmount() + sawmill.getProduction())
```
Comment (bartoszbialecki) [inline src/app/game.ts]: And here this.getWoodAmount()

File: src/app/models/buildings/building.ts [hunk 3]
```diff
@@ -55,10 +55,13 @@ export class Building {
         this.level++
 
         this.game.handleBuildingWasBuilt(this)
+        this.handleBuildingWasBuilt()
       }
     }
   }
 
+  protected handleBuildingWasBuilt() {}
```
Comment (pdybowski) [inline src/app/models/buildings/building.ts:63]: We have this function here and in extended classes by this one we create this function one More time. Do we need to keep this method? @bartoszbialecki @hoolek77

File: src/app/models/buildings/goldmine.ts [hunk 4]
```diff
@@ -34,6 +35,19 @@ export class Goldmine extends Building {
     )
   }
 
+  handleBuildingWasBuilt() {
+    if (this.passiveIncomeInterval != undefined) {
+      return
+    }
+    this.passiveIncomeInterval = setInterval(() => {
```
Comment (pdybowski) [inline src/app/models/buildings/goldmine.ts:42]: Does it work? We declare this.interval after if statement with return so won't it be always undefined and we will never have this interval?

**Review by pdybowski — COMMENTED** [review]
Dajcie odpowiedz pod komentarzami @bartoszbialecki @hoolek77

File: src/app/models/buildings/building.ts (hunk 3 above)
Comment (bartoszbialecki) [reply src/app/models/buildings/building.ts:63]: I am not sure if I understand the question, but if you want in the Building class to call this function and in subclasses to overwrite this function then I think we need it here.

File: src/app/models/buildings/goldmine.ts (hunk 4 above)
Comment (bartoszbialecki) [reply src/app/models/buildings/goldmine.ts:42]: I am not sure what you mean. First time interval is not set so you go to the block after if statement and interval is set and will be started. Next time this function is called, the interval is set so it return immediately. I think it's ok. @hoolek77 did you test it, is it ok?

File: src/app/models/buildings/building.ts (hunk 3 above)
Comment (pdybowski) [reply src/app/models/buildings/building.ts:63]: Ok, i thought it was created and later forgotten to remove

**Review by pdybowski — APPROVED** [review]

File: src/app/models/buildings/goldmine.ts (hunk 4 above)
Comment (pdybowski) [reply src/app/models/buildings/goldmine.ts:42]: Oh, its not equal.. I need to drink coffee before checking PRs 😅

**Review by bartoszbialecki — APPROVED** [review]

**Pull request merged**